library/Icinga/Web/StyleSheet.php

```diff
@@ -137,17 +144,20 @@ protected function collect()
             }
         }
 
-        if ($theme && $theme !== self::DEFAULT_THEME) {
-            if (($pos = strpos($theme, '/')) !== false) {
-                $moduleName = substr($theme, 0, $pos);
-                $theme = substr($theme, $pos + 1);
-                if ($mm->hasLoaded($moduleName)) {
-                    $module = $mm->getModule($moduleName);
-                    $this->lessCompiler->setTheme($module->getCssDir() . '/themes/' . $theme . '.less');
-                }
-            } else {
-                $this->lessCompiler->setTheme($this->pubPath . '/css/themes/' . $theme . '.less');
+        if ($themePath = self::getThemeFile($theme)) {
+                $this->lessCompiler->setTheme($themePath);
+        }
+
+        if ($user = Auth::getInstance()->getUser()) {
+            $mode = $user->getPreferences()->getValue('icingaweb', 'mode', 'none');
+
+            if (($theme === null || $theme === self::DEFAULT_THEME) && $mode === 'none') {
+                $mode = self::DEFAULT_MODE;
             }
+
+            $this->lessCompiler->addLessFile($this->pubPath . '/css/modes/'. $mode . '.less');
```

Comment:
This will never serve the file for mode `none`. You need to check the active theme here as well, in order to decide whether to respect the user's preference or apply `none` as mode if the chosen theme has no support for it.